server/fleet/microsoft_mdm.go

```diff
@@ -820,77 +823,373 @@ func (e MDMWindowsEnrolledDevice) AuthzType() string {
+	if (msg.SyncHdr.Target != nil) &&
+		(msg.SyncHdr.Target.LocURI != nil) &&
+		strings.TrimSpace(*msg.SyncHdr.Target.LocURI) != "" {
+
+		return *msg.SyncHdr.Target.LocURI, nil
+	}
+
+	return "", errors.New("message target is empty")
+}
+
+// AppendAddCommand appends a SyncML command to the Raw command list
+func (msg *SyncML) AppendCommand(cmd *SyncMLCmd) {
+	if msg.SyncBody.Raw == nil {
+		msg.SyncBody.Raw = &[]SyncMLCmd{}
+	}
+
+	if cmd != nil {
+		*msg.SyncBody.Raw = append(*msg.SyncBody.Raw, *cmd)
+	}
+}
+
+// AppendAddCommand appends a SyncML command to the Add command list
+func (msg *SyncML) AppendAddCommand(cmd SyncMLCmd) {
+	if msg.SyncBody.Add == nil {
+		msg.SyncBody.Add = &[]SyncMLCmd{}
+	}
+
+	*msg.SyncBody.Add = append(*msg.SyncBody.Add, cmd)
+}
+
+// AppendAlertCommand appends a SyncML command to the Alert command list
+func (msg *SyncML) AppendAlertCommand(cmd SyncMLCmd) {
+	if msg.SyncBody.Alert == nil {
+		msg.SyncBody.Alert = &[]SyncMLCmd{}
+	}
+
+	*msg.SyncBody.Alert = append(*msg.SyncBody.Alert, cmd)
+}
+
+// AppendAtomicCommand appends a SyncML command to the Atomic command list
+func (msg *SyncML) AppendAtomicCommand(cmd SyncMLCmd) {
+	if msg.SyncBody.Atomic == nil {
+		msg.SyncBody.Atomic = &[]SyncMLCmd{}
+	}
+
+	*msg.SyncBody.Atomic = append(*msg.SyncBody.Atomic, cmd)
+}
+
+// AppendDeleteCommand appends a SyncML command to the Delete command list
+func (msg *SyncML) AppendDeleteCommand(cmd SyncMLCmd) {
+	if msg.SyncBody.Delete == nil {
+		msg.SyncBody.Delete = &[]SyncMLCmd{}
+	}
+
+	*msg.SyncBody.Delete = append(*msg.SyncBody.Delete, cmd)
+}
+
+// AppendExecCommand appends a SyncML command to the Exec command list
+func (msg *SyncML) AppendExecCommand(cmd SyncMLCmd) {
+	if msg.SyncBody.Exec == nil {
+		msg.SyncBody.Exec = &[]SyncMLCmd{}
+	}
+
+	*msg.SyncBody.Exec = append(*msg.SyncBody.Exec, cmd)
+}
+
+// AppendGetCommand appends a SyncML command to the Get command list
+func (msg *SyncML) AppendGetCommand(cmd SyncMLCmd) {
+	if msg.SyncBody.Get == nil {
+		msg.SyncBody.Get = &[]SyncMLCmd{}
+	}
+
+	*msg.SyncBody.Get = append(*msg.SyncBody.Get, cmd)
+}
+
+// AppendReplaceCommand appends a SyncML command to the Replace command list
+func (msg *SyncML) AppendReplaceCommand(cmd SyncMLCmd) {
+	if msg.SyncBody.Replace == nil {
+		msg.SyncBody.Replace = &[]SyncMLCmd{}
+	}
+
+	*msg.SyncBody.Replace = append(*msg.SyncBody.Replace, cmd)
+}
+
+// AppendResultsCommand appends a SyncML command to the Results command list
+func (msg *SyncML) AppendResultsCommand(cmd SyncMLCmd) {
+	if msg.SyncBody.Results == nil {
+		msg.SyncBody.Results = &[]SyncMLCmd{}
+	}
+
+	*msg.SyncBody.Results = append(*msg.SyncBody.Results, cmd)
+}
+
+// AppendStatusCommand appends a SyncML command to the Status command list
+func (msg *SyncML) AppendStatusCommand(cmd SyncMLCmd) {
+	if msg.SyncBody.Status == nil {
+		msg.SyncBody.Status = &[]SyncMLCmd{}
+	}
+
+	*msg.SyncBody.Status = append(*msg.SyncBody.Status, cmd)
+}
```

Comment:
I _think_ all those `AppendXxxCommand` methods could be replaced by a generic function, although I'm not sure if a _method_ can be generic (it could possibly be turned into a helper generic function). Definitely not critical, though.

Reply:
I agree, I refactored this logic to just have a generic function.